tests/components/bayesian/test_binary_sensor.py

```diff
@@ -436,83 +493,171 @@ async def test_multiple_observations(hass: HomeAssistant) -> None:
+                    "prob_given_true": 0.2,
+                    "prob_given_false": 0.35,
+                },
+                {
+                    "platform": "numeric_state",
+                    "entity_id": "sensor.test_temp",
+                    "below": 25,
+                    "above": 15,
+                    "prob_given_true": 0.5,
+                    "prob_given_false": 0.15,
+                },
+                {
+                    "platform": "numeric_state",
+                    "entity_id": "sensor.test_temp",
+                    "above": 25,
+                    "prob_given_true": 0.15,
+                    "prob_given_false": 0.05,
                 },
             ],
-            "prior": 0.1,
+            "prior": 0.3,
         }
     }
-
     assert await async_setup_component(hass, "binary_sensor", config)
     await hass.async_block_till_done()
 
-    hass.states.async_set("sensor.test_monitored", STATE_UNKNOWN)
+    hass.states.async_set("sensor.test_temp", -5)
     await hass.async_block_till_done()
 
-    state = hass.states.get("binary_sensor.test_binary")
+    state = hass.states.get("binary_sensor.nice_day")
 
     for attrs in state.attributes.values():
         json.dumps(attrs)
-    assert state.attributes.get("occurred_observation_entities") == []
+    assert state.attributes.get("occurred_observation_entities") == ["sensor.test_temp"]
     assert state.attributes.get("probability") == 0.1
-    # No observations made so probability should be the prior
 
+    # No observations made so probability should be the prior
+    assert state.attributes.get("occurred_observation_entities") == ["sensor.test_temp"]
+    assert abs(state.attributes.get("probability") - 0.09677) < 0.01
+    # A = binary_sensor.nice_day being TRUE
+    # B = sensor.test_temp in the range [5, 10]
+    # Bayes theorum  is P(A|B) = P(B|A) * P(A) / P(B|A)*P(A) + P(B|~A)*P(~A).
+    # Where P(B|A) is prob_given_true and P(B|~A) is prob_given_false
+    # Calculated using P(A) = 0.3, P(B|A) = 0.05, P(B|~A) = 0.2 -> 0.09677
     assert state.state == "off"
 
-    hass.states.async_set("sensor.test_monitored", 20)
+    hass.states.async_set("sensor.test_temp", 5)
     await hass.async_block_till_done()
 
-    state = hass.states.get("binary_sensor.test_binary")
+    state = hass.states.get("binary_sensor.nice_day")
 
-    assert state.attributes.get("occurred_observation_entities") == [
-        "sensor.test_monitored"
-    ]
-    assert round(abs(0.026 - state.attributes.get("probability")), 7) < 0.01
-    # Step 1 Calculated where P(A) = 0.1, P(~B|A) = 0.6 (negative obs), P(~B|notA) = 0.9999 -> 0.0625
-    # Step 2 P(A) = 0.0625, P(B|A) = 0.4 (negative obs), P(B|notA) = 0.9999 -> 0.26
+    assert state.attributes.get("occurred_observation_entities") == ["sensor.test_temp"]
+    assert abs(state.attributes.get("probability") - 0.14634146) < 0.01
+    # Calculated using P(A) = 0.3, P(B|A) = 0.1, P(B|~A) = 0.25 -> 0.14634146
```

Comment:
please express the all formula and variables like in line 575

Reply:
resolved in next commit